config/environments/production.rb

```diff
@@ -18,7 +18,14 @@
   config.lograge.logger = ActiveSupport::Logger.new \
     "#{Rails.root}/log/logstash_#{Rails.env}.json"
 
-  config.staff_url = ENV.fetch('STAFF_SERVICE_URL')
+  config.staff_url = if ENV['HEROKU_APP_NAME']
```

Comment:
you also need to parse this with `URI.parse` so you can use the `#hostname` below